### DISCUSSION TITLE
Help understanding function size

### DISCUSSION THREAD

**Discussion by Falci** [vercel/next.js#53658] [Help]
### Summary

We had to update a dependency in our project which lead to a lot of code refactor. After the code changes are done, we can't deploy the project anymore due to max size limits.

> The Serverless Function "_not-found" is 50.63mb which exceeds the maximum size limit of 50mb.

`/app/not-found.ts` is very small, even considering the imports

```ts
"use client";

import { ErrorPage } from "@ui/pages";
import { UserLayout } from "@ui/layouts";

export default function Error() {
  return (
    <UserLayout>
      <ErrorPage code={404} />
    </UserLayout>
  );
}
```

I tried to remove the `/ui/index.js`, just in case there was some sort of global import (with no tree-shake)

Also, in the build log there are several blocks like this:
```
Serverless Function's page: _not-found.js
--
10:24:32.849 | Large Dependencies                     Uncompressed size  Compressed size
10:24:32.849 | node_modules/@swc/core-linux-x64-musl           51.76 MB         17.18 MB
10:24:32.849 | node_modules/@swc/core-linux-x64-gnu            44.51 MB         14.63 MB
10:24:32.849 | node_modules/next/dist                          21.99 MB          5.44 MB
10:24:32.849 | .next/server/chunks                             21.26 MB          4.57 MB
10:24:32.849 | node_modules/@esbuild/linux-x64                  8.86 MB          3.72 MB
10:24:32.849 | node_modules/webpack/lib                         3.45 MB        900.29 KB
10:24:32.849 | node_modules/sass/sass.dart.js                   4.33 MB        680.55 KB
10:24:32.849 | node_modules/react-dom/cjs                       1.64 MB        404.02 KB
10:24:32.849 | node_modules/caniuse-lite/data                  904.9 KB        325.01 KB
10:24:32.849 | node_modules/rollup/dist                       1010.3 KB         217.1 KB
10:24:32.849 | node_modules/terser/dist                       962.06 KB        184.11 KB
10:24:32.849 | node_modules/uglify-js/lib                     972.25 KB        166.19 KB
10:24:32.849 | node_modules/webpack/schemas                   611.36 KB         94.52 KB
10:24:32.850 |  
10:24:32.850 | All dependencies                               166.65 MB         49.61 MB
```

But I don't have control of most of those.
`yarn why @swc/core-linux-x64-musl` says it is used by storybook.
I moved storybook and some other deps to `devDependencies`, but it didn't change the build size

Also, I tried to move some pages that don't really SSR (like `/admin` section) to a SPA (with React Router), no success.

What else can I try?

**Comment by Falci** [comment]
I tried to update the `/app/not-found.ts` and _slight_ reduce its size to this:

```ts
export default function Error() {
  return <div>404 Not found.</div>;
}
```

and still got the "over 50mb" error

**Comment by icyJoseph** [comment]
How does the root layout look like? 

> [Bundling Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions#bundling-serverless-functions)
In order to optimize resources, there is an internal process that bundles as many Serverless Functions as possible into a single Lambda. Such a feature is currently only enabled for Next.js, but it will be enabled in other scenarios in the future.

Perhaps there's some JSON, or perhaps other kind of dependencies that are inflating the bundle size?

**Comment by Falci** [comment]
Ok, long story short: I solve it with `yarn remove storybook`

Which brings the questions: 
* Why a dev dependency affects the production build that much?
* Is it a Storybook issue or Vercel's?

**Reply by icyJoseph** [reply]
- Really? Was it used anywhere? Was it devDependency? 

Maybe it got accidentally imported into the client/server bundles?
